### ISSUE TITLE
feat(py): Add ReduceMean translator (enables LayerNorm subgraph translation)

### ISSUE THREAD

**Issue by davidrsch** [davidrsch/orbital-py-fork#14]
## Summary

`ReduceMean` computes the mean over one or more axes of a tensor. It is emitted by PyTorch and keras3 for **Layer Normalization** (`nn.LayerNorm` / `keras.layers.LayerNormalization`), which is ubiquitous in transformer-based feature extractors and modern tabular neural networks.

A typical LayerNorm subgraph in ONNX looks like:
```
ReduceMean(x)  → Sub(x, mean)
↓                     ↓
Pow(dev, 2)  → ReduceMean(sq_dev) → Add(var, eps) → Sqrt → Div → Mul(scale) → Add(bias)
```

Once `ReduceMean` is implemented (together with `Pow`, `Sqrt` from issue #14), this entire subgraph translates automatically via existing translators.

## Acceptance Criteria

- `ReduceMeanTranslator` implemented in `src/orbital/translation/steps/reducemean.py`
- For a `VariablesGroup` input with N columns and `axes=[-1]` (feature axis — the relevant case for LayerNorm), computes `(col_0 + col_1 + ... + col_N) / N` as a scalar ibis expression
- Result is broadcast back to a `ValueVariablesGroup` of N identical columns (replicated scalar mean) so that downstream `Sub(x, mean)` works element-wise
- Raises a clear `NotImplementedError` for multi-axis or batch-axis reductions (not SQL-translatable in a row-at-a-time model)
- Registered as `"ReduceMean"` in `TRANSLATORS`
- Unit tests: registered, 3-feature mean correctness

## Notes

- `keepdims` attribute: when `1` (the default), the output shape keeps the reduced dimension as size 1 — for SQL purposes this means replicate the scalar mean to all columns
- `noop_with_empty_axes`: if `axes` is empty, reduce over all axes (raise NotImplementedError)
- After this + #14 are merged, a hand-crafted ONNX LayerNorm subgraph should translate end-to-end — add an integration test

**Comment by davidrsch** [comment]
Implemented in commit `199848b` on `feat/ann-mlp-support`.\n\n- `ReduceMeanTranslator` added in `src/orbital/translation/steps/reducemean.py`\n- Handles `axes=[-1]` (feature axis) for row-at-a-time LayerNorm use case: computes `(col_0 + col_1 + ... + col_N) / N`\n- Result broadcast back to a `ValueVariablesGroup` of N identical columns so downstream `Sub(x, mean)` works\n- `keepdims=1` is the default (replicate scalar mean)\n- Raises `NotImplementedError` for batch-axis or multi-axis reductions\n- Registered as `\"ReduceMean\"` in `TRANSLATORS`\n- Unit tests: registered + 3-feature mean correctness + keepdims=0 error\n\nWith `Pow` and `Sqrt` from #16 also merged, LayerNorm subgraphs now translate end-to-end automatically.
